x-pack/plugins/actions/server/builtin_action_types/swimlane/service.ts

```diff
@@ -0,0 +1,148 @@
+ */
+
+import { Logger } from '@kbn/logging';
+import axios from 'axios';
+import https from 'https';
+import { ProxySettings } from '../../types';
+import {
+  SwimlanePublicConfigurationType,
+  ExternalService,
+  ExternalServiceCredentials,
+  CreateRecordParams,
+  CreateRecordResponse,
+  SwimlaneSecretConfigurationType,
+  MappingConfigType,
+  SwimlaneRecordPayload,
+  GetApplicationResponse,
+} from './types';
+import * as i18n from './translations';
+import { getErrorMessage, request } from '../lib/axios_utils';
+
+export const createExternalService = (
+  { config, secrets }: ExternalServiceCredentials,
+  logger: Logger,
+  proxySettings?: ProxySettings
+): ExternalService => {
+  const { apiUrl: url, appId, mappings } = config as SwimlanePublicConfigurationType;
+  const { apiToken } = secrets as SwimlaneSecretConfigurationType;
+
+  const axiosInstance = axios.create({
+    httpsAgent: new https.Agent({
+      rejectUnauthorized: false,
+    }),
+  });
+
+  if (!url || !appId || !apiToken || !mappings) {
+    throw Error(`[Action]${i18n.NAME}: Wrong configuration.`);
+  }
+
+  const headers: Record<string, string> = {
+    'Content-Type': 'application/json',
+    'Private-Token': `${secrets.apiToken}`,
+  };
+
+  const urlWithoutTrailingSlash = url.endsWith('/') ? url.slice(0, -1) : url;
+  const apiUrl = urlWithoutTrailingSlash.endsWith('api')
+    ? urlWithoutTrailingSlash
+    : urlWithoutTrailingSlash + '/api';
+  const applicationUrl = `${apiUrl}/app/{appId}`;
+  const recordUrl = `${apiUrl}/app/{appId}/record`;
+  const getApplicationUrl = (id: string) => applicationUrl.replace('{appId}', id);
+  const getPostRecordUrl = (id: string) => recordUrl.replace('{appId}', id);
+
+  const normalizeApplication = (application: {
+    id: string;
+    fields: Array<{ id: string; key: string }>;
+  }) => ({
+    id: application.id,
+    fields: application.fields,
+  });
+
+  const getBodyForEventAction = (
+    applicationId: string,
+    mappingConfig: MappingConfigType,
+    params: CreateRecordParams
+  ): SwimlaneRecordPayload => {
+    const data: SwimlaneRecordPayload = {
+      applicationId: appId,
+    };
+
+    const values: Record<string, unknown> = {};
+
+    for (const mappingsKey in mappingConfig) {
+      if (!{}.hasOwnProperty.call(mappingConfig, mappingsKey)) {
+        continue;
+      }
+
+      const fieldMap = mappingConfig[mappingsKey];
+
+      if (!fieldMap) {
+        continue;
+      }
+
+      const { id, key } = fieldMap;
+      const paramName = key.replace('KeyName', '');
+
+      values[id] = params[paramName];
+    }
+
+    data.values = values;
+
+    return data;
+  };
+
+  const getApplication = async (): Promise<GetApplicationResponse> => {
+    try {
+      const res = await request({
+        axios: axiosInstance,
+        url: getApplicationUrl(appId),
+        logger,
+        proxySettings,
+        headers,
+      });
+
+      return normalizeApplication(res.data ?? {});
```

Comment:
nit -> To avoid the function `normalizeApplication`
```suggestion
      return { ...res.?data?.application ?? {} }
```